web/src/client/software.js

```diff
@@ -52,6 +64,29 @@ const REGISTRATION_IFACE = "org.opensuse.Agama1.Registration";
  * @property {string} message - Result message.
  */
 
+/**
+ * @typedef {object} SoftwareProposal
+ * @property {string} size - Used space in human-readable form.
+ * @property {Object.<string, number>} patterns - Selected patterns and the reason.
+ */
+
+/**
+ * @typedef {object} SoftwareConfig
+ * @property {Array<string>|undefined} patterns - An object where the keys are the pattern names
```

Comment:
NP: something is wrong there. Type is an Array, but description mentions an object.